doc/N-API Basic Types.md

```diff
@@ -0,0 +1,373 @@
+# N-API Documentation
```

Comment:
We might want to put a big "WORK IN PROGRESS, NOT YET COMPLETE" at the top just to set peoples expectations.